test/integration/consul-container/libs/service/gateway.go

```diff
@@ -18,8 +18,8 @@ import (
 	"github.com/hashicorp/consul/test/integration/consul-container/libs/utils"
 )
 
-// gatewayContainer
-type gatewayContainer struct {
+// GatewayContainer
+type GatewayContainer struct {
```

Comment:
Why are we exposing `GatewayContainer`, given it satisfies the `Service` interface?

Reply:
Only `GatewayContainer` and `ConnectContainer `should have functions `getAdminAddr` given that it means envoy admin address and non-connect containers should not call this function. Removing `GetAdminAddr` from Service interface, I then have to do a type assertion before calling `GetAdminAddr`. That's why I exposed `GatewayContainer`

Reply:
Got it. But we can add empty function for the other types, which I remember is `exampleContainer` so that we can reduce the size of this PR. Additionally, I think the main purpose of using the `type Service interface ` is to not exposing these internal structs.

Reply:
My original intention with the `Service` interface was to be the lowest common denominator of anything that could be part of the consul catalog.  Concrete types would have additional methods applicable to that type as needed. We can certainly change it based on consensus, but I'm +1 smaller interfaces where everyone implements the behavior.